src/encode.c

```diff
@@ -959,6 +959,10 @@ PyImaging_LibTiffEncoderNew(PyObject *self, PyObject *args) {
                 status = ImagingLibTiffSetField(
                     &encoder->state, (ttag_t)key_int, (FLOAT64)PyFloat_AsDouble(value)
                 );
+            } else if (type == TIFF_LONG8) {
+                status = ImagingLibTiffSetField(
+                    &encoder->state, (ttag_t)key_int, PyLong_AsLongLong(value)
```

Comment:
perhaps the return value of `PyLong_AsLongLong(value)` should also be force-casted like (almost) all the others, to make it obvious what type is expected (and make it easier to find wrong modifications of the code in the future).

Reply:
Ok, I've added a commit.

Reply:
I've added a commit to cast to `uint64_t`

Reply:
Thanks, looks good.